### PR TITLE
Pagination and Ordering

### PR DESCRIPTION
**Pull request by hxoliverrumbelow**
1. Add support for DB level ordering
2. Add pagination support
3. Fix querying on a list of `OR` conditions

This PR brings us up-to-date with the new release of jsonapi-server.

### CODE REVIEW

**Comment by hxoliverrumbelow** [comment]
The build won't fail until we publish the latest version of `jsonapi-server`. Check back soon.

File: lib/sqlHandler.js [hunk 1]
```diff
@@ -311,22 +338,50 @@ SqlStore.prototype._clearAndSetRelationTables = function(theResource, partialRes
   async.parallel(tasks, callback);
 };
 
+SqlStore.prototype._generateSearchOrdering = function(request) {
+  if (!request.params.sort) return undefined;
+
+  var attribute = request.params.sort;
+  var ascending = "ASC";
```
Comment (DuncanFenning) [inline lib/sqlHandler.js]: should this be order, rather than ascending?
Assigning ascending to 'DESC' seems a bit odd

File: lib/sqlHandler.js [hunk 2]
```diff
@@ -311,22 +338,50 @@ SqlStore.prototype._clearAndSetRelationTables = function(theResource, partialRes
...
-    limit: undefined,
-    offset: undefined
-  }), function(err, result) {
+    include: self._filterInclude(request.params.filter),
+    order: self._generateSearchOrdering(request)
+  };
+  var query = _.assign(base, self._generateSearchPagination(request));
+
+  breakPromise(self.baseModel.findAndCount(query), function(err, result) {
+    debug(query, err, JSON.stringify(result));
```
Comment (DuncanFenning) [inline lib/sqlHandler.js]: Do we want debugging on as default here?

Comment (hxoliverrumbelow) [inline lib/sqlHandler.js]: The `debug` function is a no-op depending on how the process is invoked. This is the correct usage :)

Comment (DuncanFenning) [inline lib/sqlHandler.js]: Makes sense :)

**Comment by DuncanFenning** [comment]
Nice :+1:

File: lib/sqlHandler.js [hunk 3]
```diff
@@ -191,19 +196,38 @@ SqlStore.prototype._errorHandler = function(e, callback) {
...
+      where: { id: matchingValue }
     };
   });
+
+  return includeBlock;
 };
 
 SqlStore.prototype._generateSearchBlock = function(request) {
-  return this._recurseOverSearchBlock(request.params.filter);
+  var self = this;
```
Comment (pmcnr-hx) [inline lib/sqlHandler.js]: We don't really need `self` here, do we? Same closure so `self` in `228` could be `this`, right?

File: lib/sqlHandler.js [hunk 4]
```diff
@@ -311,22 +338,50 @@ SqlStore.prototype._clearAndSetRelationTables = function(theResource, partialRes
   async.parallel(tasks, callback);
 };
 
+SqlStore.prototype._generateSearchOrdering = function(request) {
+  if (!request.params.sort) return undefined;
+
+  var attribute = request.params.sort;
+  var order = "ASC";
+  attribute = ("" + attribute);
```
Comment (pmcnr-hx) [inline lib/sqlHandler.js]: Can we just use `String(attribute)` here? I still find `"" +` is not the most explicit of idioms.

**Comment by pmcnr-hx** [comment]
Quality! Just nano quibbles which aren't blockers: :+1: :shipit: :cake:

File: lib/sqlHandler.js (hunk 3 above)
Comment (hxoliverrumbelow) [inline lib/sqlHandler.js]: I'm going for consistency throughout - `self` is always my goto variable name, theres no other mingling of `this` + `self`. I'll fix up the `this` you found below :+1:

Comment (pmcnr-hx) [inline lib/sqlHandler.js]: :+1: as long as it is consistent.

**Pull request merged**